src/rviz/visualization_manager.cpp

```diff
@@ -487,62 +512,134 @@ void VisualizationManager::onUpdate()
       should_render = true;
     }
   }
-  
   if (should_render){
     render_requested_ = 0;
-    boost::mutex::scoped_lock lock(private_->render_mutex_);
     ogre_root_->renderOneFrame();
 
     if (shouldDump  && dump_images_config_->bagDuration > 0)
     {
       QPixmap screenshot_ = screen_->grabWindow(window_->winId());
-      QString filename;
-      filename.sprintf("%s/%08d.jpg", dump_images_config_->folder.c_str(), uint(++dumped_frame_count_));
+      if (screenshot_.width() % 2 || screenshot_.height() % 2) {
+          // ensure screenshot width are height are divisible by two
+          int w = screenshot_.width();
+          int h = screenshot_.height();
+          screenshot_ = screenshot_.copy(0, 0, w - (w%2), h - (h%2));
+      }
+      QImage img = screenshot_.toImage();
 
-      QImageWriter writer(filename);
-      if (!writer.write(screenshot_.toImage()))
-      {
-        QString error_message;
-        if (writer.error() == QImageWriter::UnsupportedFormatError)
-        {
-          QString suffix = filename.section('.', -1);
-          QString formats_string;
-          QList<QByteArray> formats = QImageWriter::supportedImageFormats();
-          formats_string = formats[0];
-          for (int i = 1; i < formats.size(); i++)
+      if ( private_->venc_ == NULL){
```

Comment:
This is never triggered -- `venc_` is not initialized to anything -- just points to uninitialized memory, so is not guaranteed to be null